### ISSUE TITLE
Why is tweet AST re-fetched on client side, even when AST is provided as prop?

### ISSUE THREAD

**Issue by jaisharx** [transitive-bullshit/react-static-tweets#33]
If the ast prop is provided to the tweet component, why does it re-fetches the ast on client side too? and since swr handles stuff like re-fetching when the user comes back after a network error or ideal state, there could be lot of unnecessary requests for same data, affecting the perf eventually. If devs do want to re-fetch on client side, for maybe keeping the data not stale (which would mostly depend on how they are building ast), we can provide a conditional prop like fetchOnClient=true when ast is provided. But default behavior shouldn't be to re-fetch the same ast on client even when ast is provided.

**Comment by transitive-bullshit** [comment]
The default behavior should be to only fetch the AST client-side if it's not in the server-side built cache of ASTs.

This is the line that handles this logic: https://github.com/transitive-bullshit/react-static-tweets/blob/master/packages/react-static-tweets/src/tweet.tsx#L23

If you're seeing something different in practice, please provide an example that we can debug together.

Thanks 🙏

**Comment by jaisharx** [comment]
Since the global [swrOptions](https://github.com/transitive-bullshit/react-static-tweets/blob/master/packages/react-static-tweets/src/tweet.tsx#L24) is passed with the fetcher function regardless ast prop is provided or not, what's happening is useSwr is making client side requests to the backend. 

Here's I've made a test branch for the covid.army app: [covid.army](https://covid-army-mjvyrx8wk-covidarmy.vercel.app/?location=bangalore&resource=ambulance)

![image](https://user-images.githubusercontent.com/55892116/120419622-6bfe7300-c380-11eb-9316-57fecedc8aa7.png)
When the user looses focus and comes back to the browser, requests are made to the backend even though I'm providing the ast prop.

**Comment by jaisharx** [comment]
Passing an empty fetcher function stop all the unnecessary client side requests.

```jsx
<TwitterContextProvider
                value={{
                  swrOptions: {
                    fetcher: () => null,
                  },
                }}
              >
</TwitterContextProvider>

```
But the default fetcher function with swrOptions should not be passed if ast prop is provided.

**Comment by rauchg** [comment]
Running into the same. I find this behavior really strange

**Comment by transitive-bullshit** [comment]
This should now be fixed in `v1.0.0`.
